### PR TITLE
Update configuration parsing logic

### LINKED ISSUE #68: Support yarpConfigurationFile for all plugins
At the moment it is just supported for the laser plugin. Probably we could centralize the logic to parse the driver parameters from the sdf in a single function, as done in gazebo-yarp-plugins.

### LINKED ISSUE #80: Add helper functions to locate yarp configuration files
The `yarpConfigurationFile` way to specify a configuration file for yarp plugins must support the specification of paths using the `model://` syntax.

https://github.com/robotology/gz-sim-yarp-plugins/issues/77#issuecomment-1936527309

How gazebo vs. gazebo classic search for models and other sdf elements: https://gazebosim.org/api/sim/8/migrationsdf.html

### PR DESCRIPTION
**Pull request by xela-95**
> [!NOTE]  
> This PR affects all plugins

With this PR every plugin can specify its yarp configuration both with `<yarpConfigurationString>` and `<yarpConfigurationFile>`.

When using `<yarpConfigurationFile>` the path to the file can be specified in the following ways:
- **URI** (preferred): using the `scheme[//authority]path[?query][#fragment] ` (see https://gazebosim.org/api/common/5/classgz_1_1common_1_1URI.html for more info);
-  **Relative path**
- **Absolute path**: this approach works but is highly discouraged since it does not work on other hosts.

Closes #68 
Closes #80

### CODE REVIEW

**Comment by xela-95** [comment]
@traversaro any idea on why the unit test with the CI fails in loading the plugins? I don't know how to debug this, since on my machine it works. Maybe on my machine gazebo finds the plugin libraries in the install directory?

**Comment by xela-95** [comment]
> @traversaro any idea on why the unit test with the CI fails in loading the plugins? I don't know how to debug this, since on my machine it works. Maybe on my machine gazebo finds the plugin libraries in the install directory?

Ok found! This test was missing his `GZ_SIM_SYSTEM_PLUGIN_PATH` environment variable so it was unable to find plugins.

**Comment by xela-95** [comment]
Now there is the Windows conda CI failing with:
> D:\a\gz-sim-yarp-plugins\gz-sim-yarp-plugins\libraries\common\ConfigurationHelpers.cpp(140): error C2679: binary '=': no operator found which takes a right-hand operand of type 'std::filesystem::path' (or there is no acceptable conversion)

**Comment by traversaro** [comment]
> **Relative path**

Relative path w.r.t. to what? 

By the way, do you think we could document the semantics of `yarpConfigurationFile` that you wrote down in this PR somewhere in the docs of the repo?

**Comment by xela-95** [comment]
> Relative path w.r.t. to what?
> 
> By the way, do you think we could document the semantics of `yarpConfigurationFile` that you wrote down in this PR somewhere in the docs of the repo?

Sorry for the ambiguity, the path is relative to the current working directory.

For example, looking at the [ForceTorqueTest](https://github.com/robotology/gz-sim-yarp-plugins/tree/main/tests/forcetorque), the robotinterface path is specified as:
https://github.com/robotology/gz-sim-yarp-plugins/blob/9617ad992f062e79212f7b02785345c9deb844d5/tests/forcetorque/model.sdf#L152

and it is actually resolved to the absolute path `/home/acroci/repos/gz-sim-yarp-plugins/build/tests/forcetorque/../../../tests/forcetorque/forcetorque_nws.xml`

I'm thinking to update the handling of the relative path by using `std::filesystem::canonical` or `std::filesystem::weakly_canonical` (see https://en.cppreference.com/w/cpp/filesystem/canonical) in order to remove `./` and `../` sections and have a "canonical" path.

Yep I will also add a section in the README specifying the alternatives to specify paths.

**Comment by traversaro** [comment]
I am not sure it is is a great feature, basically it (silently, not in a really clear way) constraints the sdf to be opened only if the simulator or the executable using the gz-sim as a library is launched in the directory it is intended to be launched. I guess we need it know for the tests, but probably we can avoid to advertise it/discourage to use it in actual models (a bit like absolute paths).

File: .github/workflows/conda-forge.yml [hunk 1]
```diff
@@ -89,7 +89,7 @@ jobs:
       run: |
         cd build
         # ImuTest is skipped due to https://github.com/robotology/gz-sim-yarp-plugins/issues/54
-        ctest -E "^LaserTest|^CameraTest|^ImuTest|^ForceTorqueTest|^ControlBoardCommonsTest" --repeat until-pass:5 --output-on-failure -C ${{ matrix.build_type }} .
+        ctest -E "^LaserTest|^CameraTest|^ImuTest|^ForceTorqueTest|^ControlBoardCommonsTest|^ConfigurationParsingTest" --repeat until-pass:5 --output-on-failure -C ${{ matrix.build_type }} .
```
Comment (traversaro) [inline .github/workflows/conda-forge.yml:92]: If we skip a test, it is a good idea to add a comment explaining why we are skipping it (such as the one already present before this command).

File: libraries/common/ConfigurationHelpers.cpp [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            return false;
+        }
+    } else
+    {
+        if (std::filesystem::exists(filename) && std::filesystem::is_regular_file(filename))
+        {
+            if (std::filesystem::path(filename).is_absolute())
+            {
+                yWarning() << "File specified with an absolute path: " << filename
+                           << ". It is recommended to use a URI or relative path.";
```
Comment (traversaro) [inline libraries/common/ConfigurationHelpers.cpp:135]:
```suggestion
                           << ". It is recommended to use a URI or relative path.";
```

As comment to the PR, I definitely would not recommend to use a path (relative to the current working directory) for specify configuration fails, as then the sdf will stop working if the sdf is loaded from a different cwd.

File: tests/commons/model_configuration_string.sdf [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+            <box>
+              <size>0.1 0.1 0.1</size>
+            </box>
+          </geometry>
+        </collision>
+      </link>
+
+      <!-- <plugin name="gzyarp::RobotInterface" filename="gz-sim-yarp-robotinterface-system">
+      <yarpRobotInterfaceConfigurationFile>../../../tests/camera/camera_nws.xml</yarpRobotInterfaceConfigurationFile>
+      </plugin> -->
```
Comment (traversaro) [inline tests/commons/model_configuration_string.sdf:223]: leftover?

**Review by traversaro — CHANGES_REQUESTED** [review]
See comments.

**Comment by xela-95** [comment]
> I am not sure it is is a great feature, basically it (silently, not in a really clear way) constraints the sdf to be opened only if the simulator or the executable using the gz-sim as a library is launched in the directory it is intended to be launched. I guess we need it know for the tests, but probably we can avoid to advertise it/discourage to use it in actual models (a bit like absolute paths).

After more carefully thinking about it, I agree with you. I mainly added the relative path feature in order to be compatible with the models already present in tests and tutorials that were using relative paths, but this could become tricky and cumbersome to debug for users.

Do you think it's better to remove this feature for now and refactor the paths in test and tutorials to use the URI? Or we should treat relative paths like absolute ones, by logging a warning for a potentially unreliable path while allowing it?

**Comment by traversaro** [comment]
> Do you think it's better to remove this feature for now and refactor the paths in test and tutorials to use the URI? Or we should treat relative paths like absolute ones, by logging a warning for a potentially unreliable path while allowing it?

I would go for the fast option, that I guess it is the second.

File: .github/workflows/conda-forge.yml (hunk 1 above)
Comment (xela-95) [reply .github/workflows/conda-forge.yml:93]: Done in e452f8a1261abcd2ef02039660be3293f9098c78

File: tests/commons/model_configuration_string.sdf (hunk 3 above)
Comment (xela-95) [reply tests/commons/model_configuration_string.sdf]: Solved in d3b990c256574b06cc92ea050db8e2def1667b7c

**Comment by xela-95** [comment]
macOS test disabled related issue: #90

File: libraries/common/ConfigurationHelpers.cpp (hunk 2 above)
Comment (xela-95) [reply libraries/common/ConfigurationHelpers.cpp]: Done in f3eed10b88ba3c155bdd84db0de80c7448e3aa03

**Comment by xela-95** [comment]
Updated README

File: README.md [hunk 4]
```diff
@@ -106,6 +106,39 @@ where `<install_location>` is the directory passed to `CMAKE_INSTALL_PREFIX` dur
...
+There are two ways to specify the Yarp configuration of a plugin:
+- `yarpConfigurationString`: it allows to directly specify the configuration in a string that must follow the [standard data representation format](https://www.yarp.it/latest/data_rep.html);
+- `yarpConfigurationFile`: it specifies the location of a [Yarp configuration file](https://www.yarp.it/latest/yarp_config_files.html).
+
+Concerning `yarpConfigurationFile`, the preferred way to specify the path to the configuration file is by using [Gazebo URIs](https://gazebosim.org/api/common/6/classgz_1_1common_1_1URI.html). A URI has the following general scheme:
+```
+scheme:[//authority]path[?query][#fragment] 
+```
+
+For example, the configurationn file for an IMU plugin can be specified as:
```
Comment (traversaro) [inline README.md:119]:
```suggestion
For example, the configuration file for an IMU plugin can be specified as:
```

**Comment by xela-95** [comment]
Merging! 🚀

**Pull request merged**